### PR TITLE
Add E2E tests

### PR DESCRIPTION
**Pull request by mchelen-gov**
add e2e tests

https://github.com/department-of-veterans-affairs/va.gov-team/issues/9214

related issue about E2E test disabling logic and docs: https://github.com/department-of-veterans-affairs/va.gov-team/issues/9531

AC: 
- [x] Test "incomplete" pathway
- [x] Test "pass" pathway
- [x] Test "fail" pathway"

### CODE REVIEW

**Comment by stephenBarrs** [comment]
Looks good to me for current app flow.

**Review by stephenBarrs — APPROVED** [review]

**Review by jhonny218 — APPROVED** [review]
LGTM so far

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js [hunk 1]
```diff
@@ -1,15 +1,113 @@
...
+  const isStaffButton = `button[value=${isStaff}]`;
+  const firstScrollDiv = 'div[name=multi-question-form-1-scroll-element]';
+
+  browser.waitForElementVisible(isStaffButton, Timeouts.normal);
+  browser.click(isStaffButton);
+  browser.waitForElementPresent(firstScrollDiv, Timeouts.normal);
+}
+
+/* tests end state when all answers are the same, e.g. yes or no
+    @params:
```
Comment (Yuffster-usds) [inline src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:31]: Liking the docs.

**Review by Yuffster-usds — COMMENTED** [review]
lgtm

**Review by Yuffster-usds — APPROVED** [review]

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js [hunk 2]
```diff
@@ -1,15 +1,113 @@
-const Timeouts = require('../../../platform/testing/e2e/timeouts');
-const E2eHelpers = require('../../../platform/testing/e2e/helpers');
+/* eslint-disable array-callback-return */
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:1]: It doesn't seem like you need to disable ESLint here.

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js [hunk 3]
```diff
@@ -1,15 +1,113 @@
...
+}
+
+module.exports = E2eHelpers.createE2eTest(browser => {
+  resetBrowser(browser);
+  browser.axeCheck('.main');
+  testIncompleteOnLoad(browser);
+
+  const yesNo = ['yes', 'no'];
+  yesNo.map(option => {
+    yesNo.map(option2 => {
```
Comment (bkjohnson) [inline src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:80]: Changing these `.map` calls to `.forEach` would avoid the ESLint error I think.

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js [hunk 4]
```diff
@@ -1,15 +1,113 @@
...
+
+function testEndStateForMixedAnswers(browser, isStaff) {
+  resetBrowser(browser);
+
+  testIsStaffButton(browser, isStaff);
 
-  client.end();
+  browser.elements('css selector', `button[value=no]`, links => {
+    for (let i = 2; i < links.value.length; i += 2) {
+      browser.pause(1000).elementIdClick(links.value[i].ELEMENT);
```
Comment (mchelen-gov) [inline src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:60]: @bkjohnson @jhonnyoddball where do these E2E tests results get run currently, and is there any guidance about how pauses will affect build time?

**Comment by mchelen-gov** [comment]
@jhonnyoddball @bkjohnson @stephenBarrs
Is there any way to name or group these tests? Right now the output looks like:
```
Running:  Begin application
 ✔ Element <body> was visible after 38 milliseconds.
 ✔ Expected element <div.covid-screener-results-incomplete> to be visible
 ✔ Element <body> was visible after 39 milliseconds.
 ✔ Element <button[value=yes]> was visible after 26 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 12 milliseconds.
 ✔ Expected element <div.covid-screener-results-fail> to be visible
 ✔ Expected element <div.covid-screener-results-pass> to not be present - element was not found
 ✔ Element <body> was visible after 24 milliseconds.
 ✔ Element <button[value=no]> was visible after 36 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 11 milliseconds.
 ✔ Expected element <div.covid-screener-results-fail> to be visible
 ✔ Expected element <div.covid-screener-results-pass> to not be present - element was not found
 ✔ Element <body> was visible after 33 milliseconds.
 ✔ Element <button[value=yes]> was visible after 23 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 11 milliseconds.
 ✔ Expected element <div.covid-screener-results-fail> to be visible
 ✔ Expected element <div.covid-screener-results-pass> to not be present - element was not found
 ✔ Element <body> was visible after 32 milliseconds.
 ✔ Element <button[value=yes]> was visible after 22 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 15 milliseconds.
 ✔ Expected element <div.covid-screener-results-pass> to be visible
 ✔ Expected element <div.covid-screener-results-fail> to not be present - element was not found
 ✔ Element <body> was visible after 22 milliseconds.
 ✔ Element <button[value=no]> was visible after 28 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 11 milliseconds.
 ✔ Expected element <div.covid-screener-results-pass> to be visible
 ✔ Expected element <div.covid-screener-results-fail> to not be present - element was not found
 ✔ Element <body> was visible after 39 milliseconds.
 ✔ Element <button[value=no]> was visible after 23 milliseconds.
 ✔ Element <div[name=multi-question-form-1-scroll-element]> was present after 9 milliseconds.
 ✔ Expected element <div.covid-screener-results-fail> to be visible
 ✔ Expected element <div.covid-screener-results-pass> to not be present - element was not found
```
If something broke a test case, it would be difficult to know what exactly is failing.

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js (hunk 4 above)
Comment (bkjohnson) [reply src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:60]: These are run as a part of our nightwatch tests which get run at the `e2e` stage in Jenkins here:

https://github.com/department-of-veterans-affairs/vets-website/blob/f29aa3a385d03479308f78c0c22c292ecae93f20/Jenkinsfile#L71-L72

I'm not aware of any specific guidance about how to use pauses.  There is some parallelization happening, so a 5 second pause in one test won't necessarily cause every other test to be slowed down by 5 seconds.

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js [hunk 5]
```diff
@@ -1,15 +1,113 @@
...
+function testEndStateForAnswer(browser, value, isStaff) {
+  resetBrowser(browser);
+  const visibleResultClass = value === 'no' ? passClass : failClass;
+  const notPresentResultClass = value === 'no' ? failClass : passClass;
+  const buttonValue = `button[value=${value}]`;
+
+  testIsStaffButton(browser, isStaff);
+
+  browser.elements('css selector', buttonValue, buttons => {
+    for (let i = 1; i < buttons.value.length; i++) {
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:44]: Maybe this?

```js
browser.elements('css selector', buttonValue, buttons => {
  buttons.value.forEach(btn => {
    browser.pause(1000).elementIdClick(btn.ELEMENT);
  });
});
```

File: src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js (hunk 4 above)
Comment (cvalarida) [reply src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js:60]: Undocumented rule of thumb: Don't use `.pause()` unless you absolutely have to. Instead, prefer `.waitForElementVisible()` (though it looks like the elements are already visible in this case?).

With the `.pause()`, it'll not only pause a lot, slowing down every CI build / test, but will also produce flaky tests. (Often, when Jenkins gets overwhelmed, the cause of e2e test failures are arbitrary `.pause()` durations not being long enough due to the increased load on the server.)

What happens if you have no `.pause()` statements in here?

**Comment by mchelen-gov** [comment]
@stephenBarrs is it ok if we close this in favor of https://github.com/department-of-veterans-affairs/vets-website/pull/12994 ?

**Pull request closed without merging**